### PR TITLE
Catalog tests

### PR DESCRIPTION
**Pull request by Skava600**
clickup task id - #861n34wxf

### CODE REVIEW

File: RecordHub.CatalogService/RecordHub.CatalogService.Api/Program.cs [hunk 1]
```diff
@@ -44,3 +44,5 @@
 app.MapControllers();
 
 app.Run();
+
+public partial class Program { }
```
Comment (Y45K0VL3V) [inline RecordHub.CatalogService/RecordHub.CatalogService.Api/Program.cs:48]: Do you need it?

File: RecordHub.CatalogService/RecordHub.CatalogService.Tests/Generators/BaseEntityGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+﻿using Bogus;
+using RecordHub.CatalogService.Domain.Entities;
+using RecordHub.CatalogService.Domain.Models;
+
+namespace RecordHub.CatalogService.Tests.Generators
+{
+    internal class BaseEntityGenerator
+    {
+
```
Comment (Y45K0VL3V) [inline RecordHub.CatalogService/RecordHub.CatalogService.Tests/Generators/BaseEntityGenerator.cs]: remove empty line

File: RecordHub.CatalogService/RecordHub.CatalogService.Tests/IntegrationTests/Helpers/CustomWebApplicationFactory.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using RecordHub.CatalogService.Infrastructure;
+using Testcontainers.PostgreSql;
+using WebMotions.Fake.Authentication.JwtBearer;
+
+namespace RecordHub.CatalogService.Tests.IntegrationTests.Helpers
+{
+    public class CustomWebApplicationFactory<TProgram>
+     : WebApplicationFactory<TProgram>, IAsyncLifetime where TProgram : class
```
Comment (Y45K0VL3V) [inline RecordHub.CatalogService/RecordHub.CatalogService.Tests/IntegrationTests/Helpers/CustomWebApplicationFactory.cs]: Missed the tab (or it's just github's trouble)

File: RecordHub.CatalogService/RecordHub.CatalogService.Tests/IntegrationTests/StylesControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            _factory = factory;
+            client = factory.CreateClient();
+
+            token = new ExpandoObject();
+            token.sub = Guid.NewGuid();
+            token.role = new[] { "sub_role", "Admin" };
+        }
+
+        [Fact]
+        public async Task CreateAsync_ReturnsOkResponse()
```
Comment (Y45K0VL3V) [inline RecordHub.CatalogService/RecordHub.CatalogService.Tests/IntegrationTests/StylesControllerTests.cs]: Test naming always the same "MethodName_StateUnderTest_ExpectedBehavior"

File: RecordHub.CatalogService/RecordHub.CatalogService.Tests/UnitTests/Services/StyleCatalogServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+            var model = _baseEntityGenerator.GenerateStyleModel();
+
+            var style = new Style
+            {
+                Id = styleId,
+                Name = model.Name,
+                Slug = model.Slug,
+                Records = new List<Record>
+                {
+                    new Record { Id = Guid.NewGuid(), Name = "Record 1",  },
```
Comment (Y45K0VL3V) [inline RecordHub.CatalogService/RecordHub.CatalogService.Tests/UnitTests/Services/StyleCatalogServiceTests.cs]: extra space at the end

**Review by Y45K0VL3V — CHANGES_REQUESTED** [review]

File: RecordHub.CatalogService/RecordHub.CatalogService.Api/Program.cs (hunk 1 above)
Comment (Skava600) [reply RecordHub.CatalogService/RecordHub.CatalogService.Api/Program.cs:48]: Yes - for CustomWebApplicationFactory, without it is not seeing Program class

**Review by Y45K0VL3V — APPROVED** [review]

**Pull request merged**